Implement the Button component

clementprdhomme opened the pull request:
This PR adds the `Button` component.

<p align="center">
<img width="184" alt="Screenshot of the combinations of style for the component" src="https://user-images.githubusercontent.com/6073968/27346569-8dc32d58-55e5-11e7-9c4a-8a738c820d8d.png">
</p>

Four combinations of styles are available. From top to bottom, this is how the buttons have been rendered:
- `<Button>Hey!</Button>`
- `<Button secondary>Hey!</Button>`
- `<Button inverse>Hey!</Button>`
- `<Button secondary inverse>Hey!</Button>`

The buttons can be either real buttons or links:
- if buttons, you can use the prop `onClick` to know when they've been clicked.
- if links, you can either pass a URL (prop `link`) or a route configuration (prop `link` with, for example, the following object `{ route: 'explore-index', params: { category: 'solutions' } }`.

agnlez commented on components/common/Button.js:
```diff
@@ -0,0 +1,73 @@
+
+  // We add the classes
+  if (props.secondary) classes.push('-secondary');
+  else classes.push('-primary');
```
Why didn't you use `classnames` here?

agnlez commented on components/common/Button.js:
```diff
@@ -0,0 +1,73 @@
+  if (props.secondary) classes.push('-secondary');
+  else classes.push('-primary');
+
+  if (props.inverse) classes.push('-inverse');
```
Here, aswell.

agnlez commented on components/common/Button.js:
```diff
@@ -0,0 +1,73 @@
+    // If the link is a string, this means that it is external
+    const isExternalLink = typeof props.link === 'string';
+
+    const linkAttributes = isExternalLink
```
I think we are using this in another, maybe it's worth it to create a utils for this.

clementprdhomme replied:
I don't think that can be extracted. It's very specific to the component (because of the props).

clementprdhomme commented:
I've updated the PR according to the comments.

I also added the styles for the disabled state. We don't have design for it so I just changed the opacity.
